csharp/Ice/async/Server.cs

```diff
@@ -13,37 +8,33 @@ public static int Main(string[] args)
 
         try
         {
-            using (Ice.Communicator communicator = Ice.Util.initialize(ref args, "config.server"))
+            using Ice.Communicator communicator = Ice.Util.initialize(ref args, "config.server");
+            if (args.Length > 0)
             {
-                if (args.Length > 0)
-                {
-                    Console.WriteLine("too many arguments");
-                    status = 1;
-                }
-                else
+                Console.WriteLine("too many arguments");
+                status = 1;
+            }
+            else
+            {
+                using var cts = new CancellationTokenSource();
+                Console.CancelKeyPress += (sender, eventArgs) =>
                 {
-                    using (var cts = new CancellationTokenSource())
-                    {
-                        Console.CancelKeyPress += (sender, eventArgs) =>
-                        {
-                            eventArgs.Cancel = true;
-                            cts.Cancel();
-                        };
+                    eventArgs.Cancel = true;
+                    cts.Cancel();
+                };
 
-                        Ice.ObjectAdapter adapter = communicator.createObjectAdapter("Hello");
-                        adapter.add(new HelloI(cts), Ice.Util.stringToIdentity("hello"));
-                        adapter.activate();
+                Ice.ObjectAdapter adapter = communicator.createObjectAdapter("Hello");
+                adapter.add(new HelloI(cts), Ice.Util.stringToIdentity("hello"));
+                adapter.activate();
 
-                        // cts is canceled by Ctrl+C or a shutdown request.
-                        // With C# 7.1 and up, you should make Main async and call: await Task.Delay(-1, cts.Token)
-                        cts.Token.WaitHandle.WaitOne();
+                // cts is canceled by Ctrl+C or a shutdown request.
```

Comment:
We should add a TODO or issue to fix this C# 7.1 is quite old.

Reply:
I opened an issue